svc-aggregation/system/addplugin.go

```diff
@@ -177,20 +177,21 @@ func (e *ExternalInterface) addPluginData(req AddResourceRequest, taskID, target
 
 		managersData[pluginContactRequest.OID] = body
 	}
-	e.SubscribeToEMB(plugin.ID, queueList)
 	// saving all plugin manager data
 	var listMembers = make([]agresponse.ListMember, 0)
 	for oid, data := range managersData {
-		dbErr := agmodel.GenericSave(updateManagerName(data, plugin.ID), "Managers", oid)
-		if err != nil {
-			errMsg := "error: while saving the plugin data with generic save: " + dbErr.Error()
+
+		dbErr := agmodel.SavePluginManagerInfo(updateManagerName(data, plugin.ID), "Managers", oid)
+		if dbErr != nil {
+			errMsg := "error: while saving the plugin data with SavePluginManagerInfo: " + dbErr.Error()
 			log.Error(errMsg)
-			return common.GeneralError(http.StatusInternalServerError, response.InternalError, errMsg, nil, taskInfo), "", nil
+			return common.GeneralError(http.StatusConflict, response.ResourceAlreadyExists, errMsg, []interface{}{"Plugin", "PluginID", plugin.ID}, taskInfo), "", nil
```

Comment:
where do we save ODIMRA UUID ? does it always happen before plugin manager ?

Reply:
ODIMRA UUID is saved to DB and added to managers when we bring up ODIMRA. In this function SavePluginManagerInfo, since we are using conn.create function, it ensures that everytime data is saved against unique key.(it first reads data against the key-uuid , if present throws error)